ledger/from-archive/src/main.rs

```diff
@@ -73,3 +60,33 @@ fn main() {
         block_index += 1;
     }
 }
+
+fn ledger_db(logger: &Logger, ledger_path: impl AsRef<Path>, transactions_fetcher: &ReqwestTransactionsFetcher) -> LedgerDB {
+    match ledger_path.as_ref().exists() {
+        true => {
+            log::info!(logger, "Opening existing ledger at {}", ledger_path.as_ref().display());
+            LedgerDB::open(ledger_path.as_ref()).expect("Could not open existing ledger")
+        }
+        false => create_ledger_db(logger, ledger_path, transactions_fetcher)
+    }
```

Comment:
I couldn't think of a better way to handle this, there is a timing issue here in that if 2 processes try to do this logic or similar at the same time then one will overwrite the other.

I don't think it's any worse than what we had before where this would just always stomp over the ledger db

Reply:
I think this is fine.  We run ledger-from-archive as a startup process.  There should not be any overlap between this and consensus or ledger-distribution.